json-rpc/src/util.rs

```diff
@@ -98,6 +96,75 @@ impl SdkLang {
     }
 }
 
+pub static SDK_VERSION_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"\b([0-3])\.(\d{1,2})\.(\d{1,2})\b").unwrap());
+pub static DEFAULT_SDK_VERSION: Lazy<SdkVersion> = Lazy::new(|| SdkVersion {
+    major: 0,
+    minor: 0,
+    patch: 0,
+});
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct SdkVersion {
+    pub major: u16,
+    pub minor: u16,
+    pub patch: u16,
+}
```

Comment:
Naive question for you, in order to limit the cardinality maybe we could track the actual versions (as integer data) instead of requiring that we convert it to string as a label?

Reply:
we could, but then afaik we’ll lose the ability to correlate them with any other metric (i.e failed calls, specific endpoint counts, etc)